core/src/main/java/hudson/Launcher.java

```diff
@@ -800,6 +800,10 @@ public Proc launch(ProcStarter ps) throws IOException {
             for ( int idx = 0 ; idx < jobCmd.length; idx++ )
             	jobCmd[idx] = jobEnv.expand(ps.commands.get(idx));
 
+            if (Functions.isWindows()) {
+                jobCmd = new ArgumentListBuilder(jobCmd).toWindowsCommand().toCommandArray();
```

Comment:
What happens if this windows box is running the POSIX subsystem - or cygwin?
at first pass it would appear that this could break those installations.